File: Data Structures & Algorithms/top-k-elements-in-list/submission-1.cpp
class Solution {
public:
    vector<int> topKFrequent(vector<int>& nums, int k) {
        map<int, int> freq;
        for(auto &num: nums){
            freq[num]++;
        }
        // no find the top k elements
        priority_queue<pair<int,int>> pq;
        for(auto &el : freq){
            pq.push({el.second, el.first});
        }
        vector<int> sol;
        int i =0;
        while(i!=k){
            sol.push_back(pq.top().second);
            pq.pop();
            i++;
        }
        return sol;
    }
};

File: Data Structures & Algorithms/duplicate-integer/submission-1.ts
class Solution {
    /**
     * @param {number[]} nums
     * @return {boolean}
     */
    hasDuplicate(nums: number[]): boolean {
        let mp = new Map<number, number>();
        for(const num of nums){
            let newCount = (mp.get(num) || 0) + 1;
            mp.set(num, newCount)
            if(newCount > 1){
                return true;
            }
        }
        return false;
    }
}
